test/contracts-network/colony-network.js

```diff
@@ -365,6 +394,22 @@ contract("Colony Network", accounts => {
       await checkErrorRevert(colonyNetwork.registerUserLabel(username2, orbitDBAddress, { from: accounts[1] }), "colony-user-label-already-owned");
     });
 
+    it("colony should not be able to register a user label for itself", async () => {
+      const { colony } = await setupRandomColony(colonyNetwork);
+
+      const latestVersion = await colonyNetwork.getCurrentColonyVersion();
+      const resolverAddress = await colonyNetwork.getColonyVersionResolver(latestVersion);
+      const resolver = await Resolver.at(resolverAddress);
+
+      const functionsNotAvailableOnColony = await FunctionsNotAvailableOnColony.new();
+      await resolver.register("registerUserLabel(string,string)", functionsNotAvailableOnColony.address);
+      const fakeColony = await FunctionsNotAvailableOnColony.at(colony.address);
+      await checkErrorRevert(fakeColony.registerUserLabel("test", orbitDBAddress), "colony-caller-must-not-be-colony");
```

Comment:
I sort of feel that if hitting this `require` involves overriding a Colony to add a new function, that `require` may not be necessary in the first place. Can end users even register new functions to their resolvers?

Reply:
By the rules of when to use `require`, `assert` and `revert` we should be using `assert`s here. The reason we're not is because `assert` [doesn't allow for a reason string](https://solidity.readthedocs.io/en/v0.5.9/units-and-global-variables.html?#error-handling) and therefore cannot be accurately tested. Additionally solidity-coverage doesn't ignore `assert` coverage which falsely takes our coverage down when we use them. Once the above two issues are resolved though I would be up for switching such instances to `assert`s.

Reply:
Do you know if there is anything in the pipeline to extend solidity-coverage to ignore asserts? That would be nice...